### PR TITLE
Document this repo

### PR DESCRIPTION
**Pull request by SamGerber-zz**
#### Context:

Paired with @omnibs !

Adds documentation to the gem. I know this documentation isn't perfect, but the goal is to get something up here, so we can edit and improve it.

#### Github Issue:
Outfoxes: #1 

#### PT Link

*redacted*

#### For Review

Please make an editing pass over the documentation.

It might be easier to work with if you clone the repo and start up a yard server by doing the following and visiting http://localhost:8808/ in your most favoritest browser.

```
cd ~
git clone git@github.com:NoRedInk/live_fixtures.git
cd live_fixtures
git checkout document
yard server --reload
```
   
#### For the Author

 - [x] CI :green_heart:
     <details><summary>(2)</summary>
        &#9900; If the build failed due to a flaky test:<br />
        &nbsp;&nbsp;&nbsp;&nbsp;&#9642; rerun the build until it passes. As a rule, we only merge passing builds.<br />
        &nbsp;&nbsp;&nbsp;&nbsp;&#9642; <a href="Tagging-Flaky-Tests.md">tag</a> any previously undocumented flaky tests<br />
        &#9900; If it includes major spec changes, particularly Capybara specs, consider running it in a timer loop on Jenkins to ensure it doesn't introduces non-determinism to the test suite.<br />
     </details>
 - [x] Prepare for review
     <details><summary>(2)</summary>
    If you paired on this PR, you may choose to merge without further review after completing the reviewer checklist. Otherwise, follow the steps below:<br />
        &#9900; Make a note if you _don't_ want the reviewer to merge the PR following review.<br />
        &#9900; Request a review if:<br />
        &nbsp;&nbsp;&nbsp;&nbsp;&#9642; There is a specific person you would like to review your PR, since they have domain knowledge or expertise that you would find helpful<br />
        &nbsp;&nbsp;&nbsp;&nbsp;&#9642; The PR is in silo and is consequently not well understood by most members of the team. In that case, you must assign the silo's owner.<br />
          Otherwise, the next available engineer will review it.
     </details>
 
 
 #### Merging this PR
 
 - [x] Double check that all reviewers' comments were addressed or acknowledged.
    - Comments about future refactoring should end up as TODOs in the code.
 - [x] Merge the PR.
 - [x] Delete the branch.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -22,7 +20,210 @@ Or install it yourself as:
...
+    module Seed::User
+      def self.from_fixtures(fixtures_directory)
+        insert_order = %w{users posts}
+
+        importer = LiveFixtures::Import.new fixtures_directory, insert_order
+        importer.import_all
+      end
+    end
+
+### Advanced Usage
```
Comment (omnibs) [inline README.md]: Maybe this should be a `##`? I got confused at first thinking `Advanced Usage` was a self-contained example.

File: README.md [hunk 2]
```diff
@@ -1,16 +1,14 @@
 # LiveFixtures
 
-Welcome to your new gem! In this directory, you'll find the files you need to be able to package up your Ruby library into a gem. Put your Ruby code in the file `lib/live_fixtures`. To experiment with that code, run `bin/console` for an interactive prompt.
-
-TODO: Delete this and the text above, and describe your gem
+ActiveRecord::Fixtures are a powerful way of populating data in a db;
+however, its strategy for handling primary keys and associations is
+UNACCEPTABLE for use with a production db. LiveFixtures works around this.
```
Comment (allymcknight) [inline README.md]: Why UNACCEPTABLE? Would love more explanation here.

File: README.md [hunk 3]
```diff
@@ -22,7 +20,210 @@ Or install it yourself as:
...
+
+    # In our fixtures/posts.yml file
+    posts_1234:
+      ...
+      user_id: 5678
+      summary: "Dev ops is cool."
+
+### References
+
+References allow fixtures to capture a model's associations, so they can be corrrectly re-established on import.
```
Comment (allymcknight) [inline README.md]: corrrectly -> correctly

File: README.md [hunk 4]
```diff
@@ -22,7 +20,210 @@ Or install it yourself as:
...
+The ActiveRecord::Fixture system first converts every instance of `bob` and
+`hello` into an integer using ActiveRecord::Fixture#identify, and then can
+save the records IN ANY ORDER and know that all foreign keys will be valid.
+
+There is a big problem with this. In a test db, each table is empty and so the
+odds of inserting a few dozen records causing a primary key collision is
+very small. However, for a production table with a hundred million rows, this
+is no longer the case! Collisions abound and db insertion fails.
+
+Also, auto-increment primary keys will continue from the LARGEST existing
```
Comment (allymcknight) [inline README.md:219]: Super well written explanation!!!

File: live_fixtures.gemspec [hunk 5]
```diff
@@ -27,4 +27,5 @@ Gem::Specification.new do |spec|
   spec.add_development_dependency "temping", "~> 3.0"
   spec.add_development_dependency "byebug"
   spec.add_development_dependency "sqlite3"
+  spec.add_development_dependency "yard"
```
Comment (allymcknight) [inline live_fixtures.gemspec:30]: Would you mind explaining what this line is doing? :)

**Review by allymcknight — APPROVED** [review]
Awesome job, Sam! I added a couple of questions, suggestions and pointed out a typo.

File: README.md (hunk 4 above)
Comment (SamGerber-zz) [reply README.md:219]: ❤️ I agree! 🤐 Don't tell, but I didn't write it, I just moved it around. Maybe it was @jleven or @raorao ?

Comment (raorao) [reply README.md:219]: Shoutout to @jleven

**Comment by omnibs** [comment]
SAM! The docs are beautiful! Just ran them here and took the liberty of committing fixes to some very small things I saw! Hope you don't mind!

You got my 👍 , I'm leaving the final say to @allymcknight =]

File: README.md (hunk 2 above)
Comment (SamGerber-zz) [reply README.md]: Great flag, thank you! I've softened the wording, included a bit more context, and provided a link to more info.

**Comment by omnibs** [comment]
Derp, just saw Ally already gave her 👍 .

We should eventually figure out how to host them, but for now this is awesome =]

File: README.md (hunk 1 above)
Comment (SamGerber-zz) [reply README.md]: You're right!

File: README.md (hunk 3 above)
Comment (SamGerber-zz) [reply README.md]: 😱  Tharrrnks!

**Review by omnibs — APPROVED** [review]

File: README.md (hunk 2 above)
Comment (SamGerber-zz) [reply README.md]: I lied! @omnibs beat me to it! Thanks, Juliano!

File: live_fixtures.gemspec (hunk 5 above)
Comment (SamGerber-zz) [reply live_fixtures.gemspec:30]: I'm not 100% sure, but here's what I think:

YARD is a gem that analyzes this gem's codebase and automatically creates documentation based on the code and its comments. [[more info]](http://www.rubydoc.info/gems/yard/file/README.md)

By adding it as a development dependency, it ensures that YARD will be available to future developers of the gem when they want to run the YARD server.

Your question reminds me that the docs should probably have people run YARD using `bundle exec`, so I'm glad you asked!

Maybe you or other people have more information about what this line does to share here.

**Comment by SamGerber-zz** [comment]
Thank you both for your review and contributions!

**Comment by ento** [comment]
> We should eventually figure out how to host them

I've seen some projects use http://www.rubydoc.info/

> RubyDoc.info is your source for open source Ruby library documentation, generating fresh docs for Gems and popular Git repositories.

oh, we're already on the site: http://www.rubydoc.info/gems/live_fixtures
we could bump the gem's version and publish to see the updated doc, or add this repo http://www.rubydoc.info/github (which probably won't update the gem's docs)

**Comment by SamGerber-zz** [comment]
Oh, Cool!

We could bump and publish if/when we merge https://github.com/NoRedInk/live_fixtures/pull/11 and update changelogs?

File: live_fixtures.gemspec (hunk 5 above)
Comment (allymcknight) [reply live_fixtures.gemspec:30]: Awesome! Thanks, Sam!

**Pull request merged**